### DISCUSSION TITLE
EditorConfig support

### DISCUSSION THREAD

**Discussion by nhedger** [biomejs/biome#570] [Ideas]
This informal discussion is meant to collect some ideas about implementing EditorConfig support in Biome. Much like Prettier does, the idea is to automatically apply some formatting rules depending on the contents of a `.editorconfig` file.

In no particular order, here's a non-exhaustive list of things I believe we should think about.

- Does an EditorConfig library exist for Rust?
- How to map EditorConfig rules to Biome formatting rules
- Define priorities when similar rules are declared in both an `.editorconfig` file and biome.json.

Feel free to add to this discussion, and hopefully, we'll have a better idea of what we should do.

**Comment by Arctomachine** [comment]
About priority, I think the best choice would be to throw error in console and/or editor notifications on initiation or at first run and highlight contradicting properties in biome.json (and .editorconfig, if there is such ability) to let user manually resolve conflict.
I thought about cases where this could happen and the only one I come up with is user mistake or inserting one of files from outside source with intention to tune it for current project.

**Comment by ematipico** [comment]
There's a library to parse the `ini` syntax.

https://docs.rs/rust-ini/latest/ini/

Another alternative is to create our own parser, but I don't think it's worth the time and resources.

**Comment by Conaclos** [comment]
> Define priorities when similar rules are declared in both an .editorconfig file and biome.json.

`biome.json` should take priority.

**Reply by ematipico** [reply]
https://prettier.io/docs/en/configuration.html#editorconfig

Here's what prettier does, in case we need some prior work.

**Comment by nicod-pc** [comment]
Prettier is intended to be used without a configuration to not discuss about the correct format but just use prettier. I used `.editorconfig` as a way to configure prettier, which was nice, as the options are quite limited and I thought it's maybe helpful as fallback. But as all developers use prettier for formatting, no one relies on the editorconfig implementation of the IDE anyway. So in my case it would be nice to init the `biome.json` with the settings from the `.editorconfig`, so Biome is closer to prettier. But there is no editorconfig implementation needed. It should be all covered by prettier and therefore probably also by Biome already.

**Comment by ematipico** [comment]
Closing, follow https://github.com/biomejs/biome/issues/1724 

Help will be much appreciated
